### PR TITLE
[Sprint 5]Feature get data personal information

### PR DESCRIPTION
**Pull request by Tuanlvt**
Task : Logic Profile
Link redmine: https://dev.framgia.com/redmine/issues/135500

### CODE REVIEW

File: app/src/main/res/layout/fragment_personal_infor.xml [hunk 1]
```diff
@@ -266,6 +270,7 @@
                     android:layout_marginLeft="@dimen/dp_10"
                     android:layout_marginStart="@dimen/dp_10"
                     android:background="@android:color/transparent"
+                    android:text="@{viewModel.mUser.position}"
                     android:textColor="@color/color_dark_gray"
                     android:textSize="17sp"
```
Comment (nguyenminhduong) [inline app/src/main/res/layout/fragment_personal_infor.xml]: hard code

File: app/src/main/res/layout/fragment_personal_infor.xml [hunk 2]
```diff
@@ -164,6 +166,7 @@
                     android:layout_marginLeft="@dimen/dp_10"
                     android:layout_marginStart="@dimen/dp_10"
                     android:background="@android:color/transparent"
+                    android:text="@{viewModel.mUser.phone}"
                     android:textColor="@color/color_dark_gray"
                     android:textSize="17sp"
```
Comment (nguyenminhduong) [inline app/src/main/res/layout/fragment_personal_infor.xml]: hard code

File: app/src/main/res/layout/fragment_personal_infor.xml [hunk 3]
```diff
@@ -113,6 +114,7 @@
                     android:background="@android:color/transparent"
                     android:ellipsize="end"
                     android:maxLines="1"
+                    android:text="@{viewModel.mUser.email}"
                     android:textColor="@color/color_dark_gray"
                     android:textSize="17sp"
```
Comment (nguyenminhduong) [inline app/src/main/res/layout/fragment_personal_infor.xml]: hard code

File: app/src/main/res/layout/fragment_personal_infor.xml [hunk 4]
```diff
@@ -60,6 +60,7 @@
                     android:background="@android:color/transparent"
                     android:ellipsize="end"
                     android:maxLines="1"
+                    android:text="@{viewModel.mUser.name}"
                     android:textColor="@color/color_dark_gray"
                     android:textSize="17sp"
```
Comment (nguyenminhduong) [inline app/src/main/res/layout/fragment_personal_infor.xml]: hard code

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt [hunk 5]
```diff
@@ -63,6 +62,15 @@ class MenuProfileFragment : BaseFragment(), MenuProfileContract.ViewModel {
     }
   }
 
+  fun checkLogin() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt]: can be private

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt [hunk 6]
```diff
@@ -29,20 +29,19 @@ class MenuProfileFragment : BaseFragment(), MenuProfileContract.ViewModel {
   @Inject
   lateinit var mUserRepository: UserRepository
   var mUser: ObservableField<User> = ObservableField()
+  var mVisible: ObservableField<Boolean> = ObservableField()
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt]: pls rename to more clearly

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt (hunk 5 above)
Comment (daolq3012) [reply app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt]: and rename this method to ``updateView()``

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt [hunk 7]
```diff
@@ -29,20 +29,19 @@ class MenuProfileFragment : BaseFragment(), MenuProfileContract.ViewModel {
   @Inject
   lateinit var mUserRepository: UserRepository
   var mUser: ObservableField<User> = ObservableField()
+  var mIsVisiableLayoutNotLogin: ObservableField<Boolean> = ObservableField()
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt]: ``mIsVisiableLayoutNotLogin`` -> ``mIsVisibleLayoutNotLoggedIn``

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt [hunk 8]
```diff
@@ -63,6 +62,15 @@ class MenuProfileFragment : BaseFragment(), MenuProfileContract.ViewModel {
     }
   }
 
+  private fun updateView() {
+    if (mUserRepository.getUserLocal() == null) {
+      mIsVisiableLayoutNotLogin.set(false)
+    } else {
+      mUser.set(mUserRepository.getUserLocal())
+      mIsVisiableLayoutNotLogin.set(true)
+    }
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt]: pls change to
```kt
val user = mUserRepository.getUserLocal()
mIsVisibleLayoutNotLoggedIn.set(user != null)
mUser.set(user)
```

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -60,5 +60,6 @@
     <string name="followed_by">Followed by</string>
     <string name="name">Name</string>
     <string name="following">Following</string>
+    <string name="please_login">You must be login in to perform this function. Please login !</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`please_login`" is not translated in "vi" (Vietnamese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.please_login` appears to be unused
 Unused resources make applications larger and slow down builds.

**Comment by daolq3012** [comment]
LGTM!

**Pull request merged**